osu.Game/Overlays/Mods/ModPanel.cs

```diff
@@ -0,0 +1,272 @@
+                                        Font = OsuFont.TorusAlternate.With(size: 18, weight: FontWeight.SemiBold),
+                                        Shear = new Vector2(-SHEAR_X, 0),
+                                        Margin = new MarginPadding
+                                        {
+                                            Left = -18 * SHEAR_X
+                                        }
+                                    },
+                                    new OsuSpriteText
+                                    {
+                                        Text = mod.Description,
+                                        Font = OsuFont.Default.With(size: 12),
+                                        RelativeSizeAxes = Axes.X,
+                                        Truncate = true,
+                                        Shear = new Vector2(-SHEAR_X, 0)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            Action = Active.Toggle;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(AudioManager audio, OsuColour colours)
+        {
+            sampleOn = audio.Samples.Get(@"UI/check-on");
+            sampleOff = audio.Samples.Get(@"UI/check-off");
+
+            activeColour = colours.ForModType(Mod.Type);
+            activeHoverColour = activeColour.Lighten(0.3f);
+        }
+
+        protected override HoverSounds CreateHoverSounds(HoverSampleSet sampleSet) => new HoverSounds(sampleSet);
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Active.BindValueChanged(_ =>
+            {
+                playStateChangeSamples();
+                UpdateState();
+            });
+
+            UpdateState();
+            FinishTransforms(true);
+        }
+
+        private void playStateChangeSamples()
+        {
+            if (Active.Value)
+                sampleOn?.Play();
+            else
+                sampleOff?.Play();
+        }
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            UpdateState();
+            return base.OnHover(e);
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            UpdateState();
+            base.OnHoverLost(e);
+        }
+
+        private double? mouseDownTime;
+
+        protected override bool OnMouseDown(MouseDownEvent e)
+        {
+            if (e.Button == MouseButton.Left)
+                mouseDownTime = Time.Current;
+            return true;
+        }
+
+        protected override void OnMouseUp(MouseUpEvent e)
+        {
+            mouseDownTime = null;
+            base.OnMouseUp(e);
+        }
+
+        protected override bool OnDragStart(DragStartEvent e)
+        {
+            mouseDownTime = null;
+            return true;
```

Comment:
This leads to unexpected behaviour as a button (dragging away and then back should still activate the button). Seems like it was added to avoid cross-talk between the animation states, but that's not really how this should be used. I have some changes which restructure things a bit to avoid this being necessary.